nad_ch/controllers/web/routes/column_maps.py

```diff
@@ -0,0 +1,147 @@
+import csv
+import io
+from flask import (
+    Blueprint,
+    current_app,
+    render_template,
+    g,
+    request,
+    abort,
+    redirect,
+    flash,
+    url_for,
+)
+from flask_login import login_required, current_user
+from nad_ch.application.use_cases.column_maps import (
+    add_column_map,
+    get_column_map,
+    get_column_maps_by_producer,
+    update_column_mapping,
+    update_column_mapping_field,
+)
+
+
+column_maps_bp = Blueprint("column_maps", __name__)
+
+
+@column_maps_bp.before_request
+def before_request():
+    g.ctx = current_app.extensions["ctx"]
+
+
+@column_maps_bp.route("/column-maps")
+@login_required
+def index():
+    try:
+        view_models = get_column_maps_by_producer(g.ctx, "New Jersey")
+        return render_template("column_maps/index.html", column_maps=view_models)
+    except ValueError:
+        abort(404)
+
+
+@column_maps_bp.route("/column-maps/create")
+@login_required
+def create():
+    if "name" not in request.args:
+        abort(404)
+
+    name = request.args.get("name")
+    return render_template("column_maps/create.html", name=name)
+
+
+@column_maps_bp.route("/column-maps", methods=["POST"])
+@login_required
+def store():
+    name = request.form.get("name")
+    if not name:
+        flash("Name is required")
+        return redirect(url_for("column_maps.create"))
+
+    if "mapping-csv-input" not in request.files:
+        flash("No file included")
+        return redirect(url_for("column_maps.create", name=name))
+
+    file = request.files["mapping-csv-input"]
+    if file.filename == "":
+        flash("No selected file")
+        return redirect(url_for("column_maps.create", name=name))
+
+    try:
+        file_content = file.read().decode("utf-8-sig")
+        stream = io.StringIO(file_content)
+        csv_reader = csv.reader(stream, dialect="excel")
+
+        headers = next(csv_reader)
+        if not headers:
+            flash("CSV file seems to be empty or invalid")
+            return redirect(url_for("column_maps.create", name=name))
+
+        csv_dict = {}
+
+        for row in csv_reader:
+            if len(row) < 2:
+                continue
+            key, value = row[:2]
+            csv_dict[key] = value
```

Comment:
I could see putting the CSV file processing in the use case layer, either as a separate function (`get_column_map_from_csv_file`) or have `add_column_map` accept a file as an argument.

Reply:
Good idea! Working on that now.